### ISSUE TITLE
`dead_code` lint does not trigger for traits at all.

### ISSUE THREAD

**Issue by oli-cosmian** [rust-lang/rust#66543]
I'd expect some kind of warning for a crate defining but not exporting and not using a trait. While it would be great to also report traits that are implemented for types but whose implementation is never used, this may be significantly harder to detect, not sure.

```rust
trait Foo {
    fn foo(&self) {}
    fn bar(&self) {}
}

```

([Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=7a9633d234a323a17ddcc6cc6ac75636))

Errors:

```
   Compiling playground v0.0.1 (/playground)
    Finished dev [unoptimized + debuginfo] target(s) in 1.10s

```

**Comment by oskgo** [comment]
Triage: This was fixed at some point. The playground example now gives a dead code warning as it should.
